src/TokenGenerator/PasswordGenerator.js

```diff
@@ -97,11 +108,17 @@ class PasswordGenerator extends AbstractTokenGenerator {
       method: 'POST',
       body: this.convertMapToFormData(parameters),
     }).then(response => {
-      if (response.status >= 400) {
-        handleBadResponse(response);
+      if (response.status < 400) {
+        return response;
       }
 
-      return response;
+      if (response.status === 400) {
+        return this._manageBadRequest(response);
```

Comment:
`_manageBadRequest` and `handleBadResponse` looks like the same to me, maybe we should rename `handleBadResponse` ? or use `handleBadResponse` inside _manageBadRequest`. This way the `_manageBadRequest` is used as a decorator for the public `handleBadResponse`

Reply:
BadRequest is the name associated to the 400 status code of a response, handleBadResponse is just a method name

Reply:
I renamed Error => ErrorFactory, renamed handleBadResponse => getHttpErrorFromResponse which now returns an error instead of throwing it.

Reply:
Does it breaks the external API ?

Reply:
Nope